can/bit_timing.py

```diff
@@ -153,39 +315,140 @@ def f_clock(self) -> int:
     @property
     def sample_point(self) -> float:
         """Sample point in percent."""
-        return 100.0 * (self.nbt - self.tseg2) / self.nbt
+        tmp = self.f_clock / self.bitrate
+        tmp2 = tmp / self.brp
+        btq = int(round(tmp2))
+        return round(self.tseg1 / btq * 10000) / 100.0
```

Comment:
These calculations seem to return wrong value according to the unit tests which in turn is based on existing tables. You shouldn't need to use integer calculation in Python.

Reply:
> These calculations seem to return wrong value according to the unit tests which in turn is based on existing tables.

Those existing tables have one missing bit of information that is needed in order to return the correct values, the sample point.. The other thing is the incorrectly matched bitrates to registers in those tables. as an example in the canalyst_ii.TIMING_DICT there is the bitrate 666000, this bitrate cannot be achieved using that interface. the bitrate of 666666 can be achieved.

another is the bitrate of 5000, this is impossible to get to, it's to slow.


> You shouldn't need to use integer calculation in Python.

elaborate on that